kv/service.go

```diff
@@ -0,0 +1,44 @@
+package kv
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/rand"
+	"github.com/influxdata/influxdb/snowflake"
+	"go.uber.org/zap"
+)
+
+var (
+	_ influxdb.UserService = (*Service)(nil)
+)
+
+// Service is the struct that influxdb services are implemented on.
+type Service struct {
+	kv     Store
+	Logger *zap.Logger
+
+	IDGenerator    influxdb.IDGenerator
+	TokenGenerator influxdb.TokenGenerator
+}
+
+// NewService returns an instance of a Service.
+func NewService(kv Store) *Service {
```

Comment:
Has this been called in main?  Also does Initialize need to be called in main?

Reply:
We need to call it in main, but there's a chicken/egg problem since the bolt kv store will open the same file as the bolt client.